src/components/Board.tsx

```diff
@@ -0,0 +1,136 @@
+import React, { useEffect, useState } from 'react'
+import Square from './Square'
+import { checkWinner, checkStale } from '../utils'
+import Confetti from 'react-dom-confetti'
+import confettiConfig from './confetti-config'
+import Restart from './Restart'
+
+//component, destructured props of type BoardTypeProps
+const Board = ({
+  gameHistory,
+  addGameHistory,
+  currentTurn,
+}: BoardTypeProps) => {
+  const [gameState, setGameState] = useState<GameState>(gameHistory[0])
+  const [isWinner, setIsWinner] = useState<boolean>(false)
+  const [isStale, setIsStale] = useState<boolean>(false)
```

Comment:
if you're feeling comfortable with useReducer, it would be better to move all this state (winner and stale) to the reducer also